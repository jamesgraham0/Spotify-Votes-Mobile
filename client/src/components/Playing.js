import React, { useEffect, useState } from 'react';
import { StyleSheet, View } from "react-native";
import service from '../utils/service';
import Player from "./Player";

const Playing = ({ currentlyPlaying }) => {
    // const [currentlyPlaying, setCurrentlyPlaying] = useState();

    // useEffect(() => {
    //     const getTrack = async () => {
    //         let track = await service.getCurrentlyPlaying();
    //         setCurrentlyPlaying(track);
    //     }
    //     getTrack();
    // }, []);

    return (
    <View style={styles.container}>
        <View style={styles.player}>
            <Player currentlyPlaying={currentlyPlaying}/>
        </View>
    </View>
  );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#191414",
        alignItems: "center",
    },
    player: {
        backgroundColor: '#080808',
        width: '100%',
        height: '100%',
    }
});

export default Playing;
