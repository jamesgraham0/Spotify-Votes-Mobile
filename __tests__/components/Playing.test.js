import React from 'react';
import renderer from 'react-test-renderer';
import Playing from '../../client/src/components/Playing';

describe('Playing', () => {
    it('should render the Playing component', () => {
        user = {
            id: 1,
            name: "John",
            email: "",
            image: "",
        }
        room = {
            id: 1,
            name: "Room",
            code: "1234",
            host: user,
            users: [user],
            queue: [],
            playing: false,
            currentSong: null,
            votes: [],
        }
        const tree = renderer.create(<Playing user={user} room={room}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});