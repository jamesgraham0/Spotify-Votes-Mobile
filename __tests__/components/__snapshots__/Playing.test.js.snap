// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playing should render the Playing component 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
      },
      {
        "backgroundColor": "#191414",
      },
    ]
  }
>
  <View
    style={
      {
        "backgroundColor": "#080808",
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
